### ISSUE TITLE
Обработка ошибок

### ISSUE THREAD

**Issue by dmitryme** [quantboxru/sbe#10]
https://github.com/quantboxru/sbe/blob/5cf4974f11a2cc3733fd8fafdf080cac9a0d0cde/solution/car.h#L60

Как нотифицировать, что произошла какая-то ошибка? Например в процедуру decode засунули некоректный буфер и т.д.

**Comment by Sapfear13** [comment]
Ну, тут выбор: либо возвращать NULL ptr, либо вместо void* можно указать int, а в параметры функции указать void* ptr. У дальше - если int !=0, то по коду можно понять что произошло + если там будет много ошибок то можно сделать битовую маску

**Comment by dmitryme** [comment]
Да, можно возвращать NULL и например передавать в decode указатель на sbe_error_t*

sbe_secode(char* buf, size_t len, sbe_error_t** err), если он не нулевой, то возвращается указатель на объект ошибки sbe_error_t*
